packages/next-swc/crates/core/src/relay.rs

```diff
@@ -102,8 +103,16 @@ fn path_for_artifact(
 
     if let Some(artifact_directory) = &config.artifact_directory {
         Ok(root_dir.join(artifact_directory).join(filename))
+    } else if real_file_name.starts_with(root_dir.join("./pages")) {
```

Comment:
The page directory could also be in `./src/pages`, I think this is already exposed under `opts.pages_dir` as used here https://github.com/vercel/next.js/blob/04ca8dbff3595a603423436d67d27cbc81849ad5/packages/next-swc/crates/core/src/lib.rs#L145